lib/evaluate/variable.h

```diff
@@ -288,7 +290,12 @@ class Substring {
   }
 
   Expr<SubscriptInteger> lower() const;
+  Expr<SubscriptInteger> *lower();
   Expr<SubscriptInteger> upper() const;
+  Expr<SubscriptInteger> *upper();
```

Comment:
In `variable.h` classes, the following accessors of optional members are returning `std::optional` instead of a pointer:

- `Triplet::upper()`
- `Triplet::lower()`
- `CoarrayRef::stat()`
- `CoarrayRef::team()`

Should theses accessors be changed to return pointer instead so that they follow the same pattern as `Substring::lower()` and `Substring::upper()`?

I did not get far enough when experimenting with folding using the mutable visitor framework, but it seems to me that returning optional is an issue for the mutable `Descend` because it ends up descending in copies of the member rather than the member themselves.

Reply:
Yes, that's a bad interface design.  I'll clean it up before merging.